internal/resource-request-operator/broadcaster.go

```diff
@@ -147,59 +181,119 @@ func (b *Broadcaster) onNodeDelete(obj interface{}) {
 }
 
 func (b *Broadcaster) onPodAdd(obj interface{}) {
-	pod := obj.(*corev1.Pod)
-	klog.V(4).Infof("OnPodAdd: Add for pod %s:%s\n", pod.Namespace, pod.Name)
-	podResources := extractPodResources(pod)
-	currentResources := b.readClusterResources()
-	// subtract the pod resource from cluster resources. This action is done for all pods to extract actual available resources.
-	subResources(currentResources, podResources)
-	b.writeClusterResources(currentResources)
-	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
-		klog.V(4).Infof("OnPodAdd: Pod %s:%s passed ClusterID check. ClusterID = %s\n", pod.Namespace, pod.Name, clusterID)
-		currentPodsResources := b.readPodResources(clusterID)
-		// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
-		// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
-		addResources(currentPodsResources, podResources)
-		b.writePodsResources(clusterID, currentPodsResources)
+	podAdded := obj.(*corev1.Pod)
+	klog.V(4).Infof("OnPodAdd: Add for pod %s:%s\n", podAdded.Namespace, podAdded.Name)
+	if pod.IsPodReady(podAdded) {
+		podResources := extractPodResources(podAdded)
+		currentResources := b.readClusterResources()
+		// subtract the pod resource from cluster resources. This action is done for all pods to extract actual available resources.
+		subResources(currentResources, podResources)
+		b.writeClusterResources(currentResources)
+		if clusterID := podAdded.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+			klog.V(4).Infof("OnPodAdd: Pod %s:%s passed ClusterID check. ClusterID = %s\n", podAdded.Namespace, podAdded.Name, clusterID)
+			currentPodsResources := b.readPodResources(clusterID)
+			// add the resource of this pod in the map clusterID => resources to be used in ReadResources() function.
+			// this action is done to correct the computation not considering pod offloaded by the cluster with this ClusterID
+			addResources(currentPodsResources, podResources)
+			b.writePodsResources(clusterID, currentPodsResources)
```

Comment:
Please be consistent, one function is readPodResources, while the other is writePod**s**Resources.